src/main/groovy/io/jenkins/infra/repository_permissions_updater/ArtifactoryPermissionsUpdater.groovy

```diff
@@ -146,18 +146,42 @@ class ArtifactoryPermissionsUpdater {
                         users [:]
                     } else {
                         users definition.developers.collectEntries { developer ->
-                            if (!KnownUsers.exists(developer.toLowerCase())) {
-                                reportChecksApiDetails(developer + " needs to login to artifactory", 
+                            def existsInArtifactory = KnownUsers.existsInArtifactory(developer)
+                            def existsInJira = KnownUsers.existsInJira(developer) || JiraAPI.getInstance().isUserPresent(developer)
+
+                            if (!existsInArtifactory && !existsInJira) {
+                                reportChecksApiDetails(developer + " needs to log in to Artifactory",
```

Comment:
hopefully & is a valid character, worried about the length as it will get cut off if it's too long, thoughts? easy to test after merge
```suggestion
                                reportChecksApiDetails(developer + " needs to log in to Artifactory & Jira",
```

Reply:
Why not sidestep the problem and go with

```suggestion
                                reportChecksApiDetails(developer + " needs to log in to Artifactory and Jira",
```

Reply:
sure go for it, let's see what it looks like in the UI after merge

Reply:
Will amend when I update the report URL.